src/ControlComponents/ModeInput.jsx

```diff
@@ -12,17 +12,36 @@ export default function ModeInput({
   clickFunction,
   header,
   inputId,
+  externalValue,
 }) {
+  const [inputField, setInputField] = useState(undefined);
+  const [currentValue, setCurrentValue] = useState(externalValue);
+
+  useEffect(() => {
+    setInputField(document.getElementById(inputId));
+    // eslint-disable-next-line
+  }, []);
+
+  useEffect(() => {
+    if (!inputField) {
+      return;
+    }
+    if (externalValue !== currentValue) {
+      setCurrentValue(externalValue);
+      inputField.value = roundNumber(externalValue);
+    }
+    // eslint-disable-next-line
+  }, [externalValue]);
+
   // Handles button click by sending value to parent
   const handleClick = () => {
-    const inputField = document.getElementById(inputId);
-    const inputValue = Number(inputField.value);
-    const prettyValue = roundNumber(inputValue);
+    setCurrentValue(inputField.value);
+    const prettyValue = roundNumber(currentValue);
```

Comment:
When testing I'm getting errors in roundNumber which is caused by this I believe. setCurrentValue doesn't update currentValue immediately so inputField.value should be used instead of currentValue.

Reply:
Good catch! I have commited a (hopefully) fix, so please try again 👷